scripts/ahk.js

```diff
@@ -467,9 +508,57 @@ function _check_form(show_error = true, check_required_fields = false) {
         }
     }
 
+    if (!result) {
+        return result;
+    }
+
+    // Shorten URL
+    if(FEATURE_TOGGLES.ENABLE_COMPRESSION) {
+        // https://stackoverflow.com/a/317000??
+
+        var user_requested_shortened = $('#chkBox_CompressData').is(':checked')
+        _debug_log("User requested shorten:", user_requested_shortened);
+        var formData = new FormData($('#hotkeyForm')[0]);
+        searchParams = new URLSearchParams(formData);
+        _debug_log("params: ", searchParams);
+        queryString = searchParams.toString();
+        _debug_log("QueryString:", queryString);
+
+        var should_shorten = user_requested_shortened;
+        var limit = 8.2e3
+        if(location.host.startsWith('localhost')) {
+            limit = 2e3
+        }
+        _debug_log(limit)
+        _debug_log("length:", (location.href + queryString).length)
+        if (!user_requested_shortened && (location.href + queryString).length > limit) {
+            _debug_log("warning that should shorten")
+            displayYesNoLinks(
+                "Shorten URL?",
+                `<p>The new configuration URL may be too long (${location.href.length + queryString.length} is 
+                greater than ${limit}).</p><p>Shorten the URL?<br/>("YES"
+                 to shorten and proceed, "NO" to proceed as is, or close this dialogue to cancel)</p>`, 
+                `/?${_get_shortened_url(queryString)}`, `/?${queryString}`, 
+                true
+            )
+            return false;
+        }
+
+        if (should_shorten) {
+            window.location.href='/?' + _get_shortened_url(queryString)
+            return false
+        }
```

Comment:
these two blocks should swap, then the user_requested_shortened is definitely false when the if is reached